features/steps/givens/entities.py

```diff
@@ -58,4 +58,10 @@ def step_impl(context, inst, relationship_direction):
     attr_value = getattr(inst, attribute_name, NullAttribute)
     if attr_value is None:
         attr_value = NullAttribute
-    yield ValidationOutcome(instance_id = attr_value, severity = OutcomeSeverity.PASSED)
\ No newline at end of file
+    yield ValidationOutcome(instance_id = attr_value, severity = OutcomeSeverity.PASSED)
+
+
+@gherkin_ifc.step("All referenced instances")
+def step_impl(context, inst):
+    instances = context.model.traverse(inst)
+    yield ValidationOutcome(instance_id=instances, severity=OutcomeSeverity.PASSED)
```

Comment:
@Ghesselink In other occasions I see:

```
for inst in instances:
        yield ValidationOutcome(instance_id = inst, severity = OutcomeSeverity.PASSED)
```

Does it matter?

Reply:
Yes, unfortunately. For reasons that are perhaps a bit complex.
In the occasions that you mention, the way we handle the outputted results from the step_impl is.
https://github.com/buildingSMART/ifc-gherkin-rules/blob/fa4b6e2f6335147fabf50813491a2fda795af80f/features/steps/validation_handling.py#L95
Note that we move the entire output list to the context stack. We do this solely to determine the initial set of activation instances. If we don't, i.e., if a statement executes a function with one of the activation instances as an argument (as is the case for any function following `Given an IfcEntity`), we move only the first element of the list to the context stack. 
https://github.com/buildingSMART/ifc-gherkin-rules/blob/fa4b6e2f6335147fabf50813491a2fda795af80f/features/steps/validation_handling.py#L61

Note the `[0]`. This works in most cases (in fact, in all cases so until now) because the output list has a length of 1, containing a single yielded entity instance (or value) resulting from step_impl. I admit it's not ideal, but it could handle the complexity of irregular nested lists. 
In case we do the same for the current statement (`for inst in instances; yield ValidationOutcome`), this [0] indicates that we will take the first element of context.model.traverse(inst). This includes inst as the first element in this list, meaning that context.instances will remain exactly the same.

Reply:
Clear. Thanks for clarifying, but indeed something to look out for.